### PR TITLE
Fix minor issues + Add tests

### PR DESCRIPTION
**Pull request by egeloen**
All are in the title :kiss:

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -31,10 +31,21 @@ public function getConfigTreeBuilder()
...
+                ->scalarNode('profile_id')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->scalarNode('private_key_file')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->scalarNode('http_adapter')
+                    ->defaultValue('curl')
```
Comment (tmartin) [inline DependencyInjection/Configuration.php]: IMO it was better to let here the full name of the service. This way, any developer could use he's own implementation of `\Widop\HttpAdapterBundle\Model\HttpAdapterInterface`.

File: Model/Client.php [hunk 2]
```diff
@@ -102,21 +85,25 @@ public function getPrivateKeyFile()
     /**
      * Sets the absolute private key file path.
      *
-     * @param string $privateKeyFile The absolute private key file path.
+     * @throws \InvalidArgumentException If the private key file does not exist.
      *
-     * @return \Widop\GoogleAnalyticsBundle\Model\Client
+     * @param string $privateKeyFile The absolute private key file path.
```
Comment (tmartin) [inline Model/Client.php:88]: I think we said `@throws` should be place after `@param`

File: Model/Client.php [hunk 3]
```diff
@@ -124,23 +111,21 @@ public function getHttpAdapter()
     }
 
     /**
-     * Sets the Http Adapter used by the client.
-     *
-     * @param \Widop\HttpAdapterBundle\Model\HttpAdapterInterface $httpAdapter The Http Adapter used by the client.
+     * Sets the http adapter.
      *
-     * @return \Widop\GoogleAnalyticsBundle\Model\Client
+     * @param \Widop\HttpAdapterBundle\Model\HttpAdapterInterfac$httpAdaptere $httpAdapter The http adapter.
```
Comment (tmartin) [inline Model/Client.php]: Typo in the doc.

File: Model/Client.php [hunk 4]
```diff
@@ -167,65 +152,61 @@ public function getAccessToken()
...
 
         $jwtSignature = base64_encode($this->generateSignature($jwtHeader.'.'.$jwtClaimSet));
 
-        return $jwtHeader.'.'.$jwtClaimSet.'.'.$jwtSignature;
+        return sprintf('%s.%s.%s', $jwtHeader, $jwtClaimSet, $jwtSignature);
     }
 
     /**
-     * Generated the jwt signature according to the private key file and the jwt content.
+     * Generated the JWT signature according to the private key file and the JWT content.
```
Comment (tmartin) [inline Model/Client.php]: Generates

File: Model/Client.php [hunk 5]
```diff
@@ -167,65 +152,61 @@ public function getAccessToken()
...
      *
-     * @return string
+     * @return string The JWT signature.
      */
-    private function generateSignature($jwt)
+    protected function generateSignature($jsonWebToken)
     {
-        if (!function_exists('openssl_x509_read')) {
-            throw new \Exception();
-        }
```
Comment (tmartin) [inline Model/Client.php:206]: Why remove this code?

File: Model/Client.php [hunk 6]
```diff
@@ -167,65 +152,61 @@ public function getAccessToken()
...
-
         $certificate = file_get_contents($this->privateKeyFile);
 
         $certificates = array();
         if (!openssl_pkcs12_read($certificate, $certificates, 'notasecret')) {
-            throw new \Exception();
+            throw new \Exception('An error occured when parsing the PKCS 12 certificate.');
```
Comment (tmartin) [inline Model/Client.php:200]: Should we create a specific exception class? What do you think?

File: Model/GoogleAnalyticsService.php [hunk 7]
```diff
@@ -47,18 +45,14 @@ public function getClient()
...
     public function setClient(Client $client)
     {
         $this->client = $client;
-
-        return $this;
     }
 
     /**
-     * Query the google analytics service.
+     * Queries the google analytics service.
```
Comment (tmartin) [inline Model/GoogleAnalyticsService.php:55]: :+1: :kiss:

File: Model/Query.php [hunk 8]
```diff
@@ -231,28 +231,22 @@ public function setDimensions(array $dimensions)
...
     }
 
     /**
-     * Checks if the google analytics query has sorts.
+     * Checks if the google analytics query is ordered.
      *
-     * @return boolean TRUE if the google analytics query has sorts else FALSE.
+     * @return boolean TRUE if the google analytics query is ordered else FALSE.
      */
     public function hasSorts()
```
Comment (tmartin) [inline Model/Query.php:251]: Why change this doc, since the method is still `hasSorts`?

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (egeloen) [inline DependencyInjection/Configuration.php]: Done

File: Model/Client.php (hunk 2 above)
Comment (egeloen) [inline Model/Client.php:88]: Fixed

File: Model/Client.php (hunk 3 above)
Comment (egeloen) [inline Model/Client.php]: Fixed

File: Model/Client.php (hunk 4 above)
Comment (egeloen) [inline Model/Client.php]: Fixed

File: Model/Client.php (hunk 5 above)
Comment (egeloen) [inline Model/Client.php:206]: Add it. Btw, how should you test this part?

File: Model/Client.php (hunk 6 above)
Comment (egeloen) [inline Model/Client.php:200]: Probably :) For now, I prefer keep it like that... I have opened #10

File: Model/Query.php (hunk 8 above)
Comment (egeloen) [inline Model/Query.php:251]: If there is `sorts` means in fact the query is ordered... This is the reason :)

File: README.md [hunk 9]
```diff
@@ -1,167 +1,25 @@
-# WidopGoogleAnalyticsBundle
+# README
 
-The `WidopGoogleAnalyticsBundle` provides a way to use certificate-based authentication in server-to-server interactions for google analytics.
+[![Build Status](https://secure.travis-ci.org/widop/WidopGoogleAnalyticsBundle.png)](http://travis-ci.org/widop/WidopGoogleAnalyticsBundle)
 
-In other words, no user interaction will be required thanks to the ".p12" certificate and you will be able to automate your google analytics authentication and authorization processes as well as the way of querying the google analytics service.
+The `WidopGoogleAnalyticsBundle` provides a way to use certificate-based authentication in server-to-server
+interactions for google analytics.
```
Comment (tmartin) [inline README.md]: s/for/with/

File: Resources/doc/usage.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+Here is an example of what you can do with the query object:
+
+``` php
+$query = $this->container->get('widop_google_analytics.query');
+$query->setIds('ga:XXXXXXXX');
+$query->setDimensions(array('ga:eventLabel'));
+$query->setStartDate(new \DateTime('2012-01-01'));
+$query->setEndDate(new \DateTime());
+$query->setMetrics(array('ga:uniqueEvents'));
+$query->setFilters(array('ga:eventCategory==travelClick'));
```
Comment (tmartin) [inline Resources/doc/usage.md:80]: Are you sure of the syntax here?

File: composer.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "widop/google-analytics-bundle",
-    "description": "Provides a way to use certificate-based authentication in server-to-server interactions for google analytics",
+    "description": "Google certificate-based authentication in server-to-server interactions for google analytics",
```
Comment (tmartin) [inline composer.json]: s/for/with/

File: README.md (hunk 9 above)
Comment (egeloen) [inline README.md]: Fixed

File: composer.json (hunk 11 above)
Comment (egeloen) [inline composer.json]: Fixed

File: Resources/doc/usage.md (hunk 10 above)
Comment (egeloen) [inline Resources/doc/usage.md:80]: Yes :)

**Pull request merged**